package/MDAnalysis/due.py

```diff
@@ -0,0 +1,73 @@
+# emacs: at the end of the file
+# ex: set sts=4 ts=4 sw=4 et:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### #
+"""
+
+Stub file for a guaranteed safe import of duecredit constructs:  if duecredit
+is not available.
+
+To use it, place it into your project codebase to be imported, e.g. copy as
+
+    cp stub.py /path/tomodule/module/due.py
+
+Note that it might be better to avoid naming it duecredit.py to avoid shadowing
+installed duecredit.
+
+Then use in your code as
+
+    from .due import due, Doi, BibTeX
+
+See  https://github.com/duecredit/duecredit/blob/master/README.md for examples.
+
+Origin:     Originally a part of the duecredit
+Copyright:  2015-2016  DueCredit developers
+License:    BSD-2
+"""
+
+__version__ = '0.0.5'
+
+
+class InactiveDueCreditCollector(object):
+    """Just a stub at the Collector which would not do anything"""
+    def _donothing(self, *args, **kwargs):
+        """Perform no good and no bad"""
+        pass
+
+    def dcite(self, *args, **kwargs):
+        """If I could cite I would"""
+        def nondecorating_decorator(func):
+            return func
+        return nondecorating_decorator
+
+    cite = load = add = _donothing
+
+    def __repr__(self):
+        return self.__class__.__name__ + '()'
+
+
+def _donothing_func(*args, **kwargs):
+    """Perform no good and no bad"""
+    pass
+
+
+try:
+    from duecredit import due, BibTeX, Doi, Url
+    if 'due' in locals() and not hasattr(due, 'cite'):
+        raise RuntimeError(
+            "Imported due lacks .cite. DueCredit is now disabled")
+except Exception as e:
+    if type(e).__name__ != 'ImportError':
+        import logging
+        logging.getLogger("duecredit").error(
```

Comment:
This error message is not good. I tried (with `duecredit` not installed) under Python 3.6
```
DUECREDIT_ENABLE=yes python -c "import MDAnalysis"
```
and got
```
Failed to import duecredit due to No module named 'duecredit'
```
I *think* this error message should not have even appeared. This comes from a `ModuleNotFoundError` as far as I can tell ,which is a subclass of `ImportError`. 

How about doing something like

```python
except ImportError:
   pass
except Exception as err:
   import logging
   logging.getLogger("duecredit").error("Failed to import duecredit due to {}", str(e))
```

Reply:
Please also address this comment.

Reply:
I have addressed other issues, but I am not sure if I get this correctly.  If anything is ran with DUECREDIT_ENABLE = true, then shouldn't it be obvious that the user has duecredit installed. I think the maximum we can do is to show a warning that either duecredit needs to be installed or the change the environment variable.  What do you think? @orbeckst  . If it is not installed and DUECREDIT_ENABLE is not set, then it wouldn't give an error.

Reply:
The original intent of the code was, if I read it correctly, to *not show a warning when the module cannot be imported* (which is what duecredit promises). What I am saying is that this intended behavior is not working because the test `if type(e).__name__ != 'ImportError':` is wrong (at least in 3.6): it does not catch the `ModuleNotFoundError` whereas just catching `ImportError` will cleanly accomplish the goal.

By the way: when I import MDAnalysis with your current code and when `duecredit` is *not installed* I get
```python
>>> import MDAnalysis
Failed to import duecredit due to No module named 'duecredit'
>>> import os
>>> os.environ['DUECREDIT_ENABLE']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/oliver/anaconda3/envs/mda_develop/lib/python3.6/os.py", line 669, in __getitem__
    raise KeyError(key) from None
KeyError: 'DUECREDIT_ENABLE'
```
i.e., most users will see this warning. The behavior needs to be that users are not bothered with this import error.